subsys/bluetooth/host/att.c

```diff
@@ -2948,6 +2989,57 @@ static int bt_att_accept(struct bt_conn *conn, struct bt_l2cap_chan **ch)
 BT_L2CAP_CHANNEL_DEFINE(att_fixed_chan, BT_L2CAP_CID_ATT, bt_att_accept, NULL);
 
 #if defined(CONFIG_BT_EATT)
+static k_timeout_t credit_based_connection_delay(struct bt_conn *conn)
+{
+	/*
+	 * 5.3 Vol 3, Part G, Section 5.4 L2CAP COLLISION MITIGATION
+	 * ... In this situation, the Central may retry
+	 * immediately but the Peripheral shall wait a minimum of 100 ms before retrying;
+	 * on LE connections, the Peripheral shall wait at least 2 *
+	 * (connPeripheralLatency + 1) * connInterval if that is longer.
+	 */
+
+	if (conn->role == BT_CONN_ROLE_CENTRAL) {
+		return K_NO_WAIT;
+	}
+
+	const uint32_t rand_delay = sys_rand32_get() & 0x7; /* Small random delay for IOP */
+	const uint32_t calculated_delay =
+		2 * (conn->le.latency + 1) * BT_CONN_INTERVAL_TO_MS(conn->le.interval);
+
+	return K_MSEC(MAX(100, calculated_delay + rand_delay));
```

Comment:
We could (and probably should) guard the two parts here with `IS_ENABLED(CONFIG_BT_CENTRAL)` and `IS_ENABLED(CONFIG_BT_PERIPHERAL)`